src/librustc/infer/error_reporting/anon_anon_conflict.rs

```diff
@@ -0,0 +1,197 @@
+                            }
+                        })
+                        .next()
+                } else {
+                    None
+                }
+            } else {
+                None
+            }
+        } else {
+            None
+        }
+    }
+
+    pub fn try_report_anon_anon_conflict(&self, error: &RegionResolutionError<'tcx>) -> bool {
+
+        let (span, sub, sup) = match *error {
+            ConcreteFailure(ref origin, sub, sup) => (origin.span(), sub, sup),
+            _ => return false, // inapplicable
+        };
+
+        // Determine whether the sub and sup consist of both anonymous (elided) regions.
+        if let (Some(br1), Some(br2)) =
+            (self.is_anonymous_region(sup), self.is_anonymous_region(sub)) {
+            if let (Some(ty1), Some(ty2)) =
+                (self.find_anon_type(sup, &br1), self.find_anon_type(sub, &br2)) {
+                if let (Some((anon_arg1, _, _, _)), Some((anon_arg2, _, _, _))) =
+                    (self.find_arg_with_anonymous_region(sup, sup),
+                     self.find_arg_with_anonymous_region(sub, sub)) {
+
+                    let span_label_var1 = if let Some(simple_name) = anon_arg1.pat.simple_name() {
+                        format!(" from `{}` ", simple_name)
+                    } else {
+                        format!(" ")
+                    };
+
+                    let span_label_var2 = if let Some(simple_name) = anon_arg2.pat.simple_name() {
+                        format!(" into `{}` ", simple_name)
+                    } else {
+                        format!(" ")
+                    };
+
+                    struct_span_err!(self.tcx.sess, span, E0623, "lifetime mismatch")
+                        .span_label(ty1.span,
+                                    format!("these references must have the same lifetime"))
+                        .span_label(ty2.span, format!(""))
+                        .span_label(span,
+                                    format!("data{}flows{}here", span_label_var1, span_label_var2))
+                        .emit();
+                    return true;
+                }
+            }
+        }
+        false
+    }
+
+    pub fn is_anonymous_region(&self, region: Region<'tcx>) -> Option<ty::BoundRegion> {
+
+        match *region {
+            ty::ReFree(ref free_region) => {
+                match free_region.bound_region {
+                    ty::BrAnon(..) => {
+                        let anonymous_region_binding_scope = free_region.scope;
+                        let node_id = self.tcx
+                            .hir
+                            .as_local_node_id(anonymous_region_binding_scope)
+                            .unwrap();
+                        match self.tcx.hir.find(node_id) {
+                            Some(hir_map::NodeItem(..)) |
+                            Some(hir_map::NodeTraitItem(..)) => {
+                                // proceed ahead //
```

Comment:
I would change this to something like: "// Success -- proceed to return `Some` below"